packages/safe-tools-client/app/components/setup-safe-modal/index.gts

```diff
@@ -0,0 +1,80 @@
+import Component from '@glimmer/component';
+import { on } from '@ember/modifier';
+
+import { svgJar } from '@cardstack/boxel/utils/svg-jar';
+import BoxelModal from '@cardstack/boxel/components/boxel/modal';
+import BoxelActionContainer from '@cardstack/boxel/components/boxel/action-container'
+
+import './index.css';
+
+interface Signature {
+  Element: HTMLElement;
+  Args: { 
+    isOpen: boolean 
+    loading?: boolean;
+    onClose: () => void;
+  };
+}
+
+//TODO: replace with correct flags and logic
+const gasCost = `0.001899365 ETH (USD$3.01)`;
+const hasEnoughGas = true;
+
+export default class SetupSafeModal extends Component<Signature> {
+  <template>
+    <BoxelModal
+      @size='medium'
+      @isOpen={{@isOpen}}
+      @onClose={{@onClose}}
+      aria-labelledby='boxel-modal-setup-safe'
+    >
+      <BoxelActionContainer as |Section ActionChin|>
+        <Section @title='Set up a Payment Safe' class='setup-safe-modal__section'>
+          <p>In this step, you create a safe equipped with a module to schedule
+            payments.
+            <br />What you need to know:
+          </p>
+          <ul>
+            <li>Your payment safe is used to fund your scheduled payments. </li>
+            <li>You are the owner of the safe. Only the safe owner can schedule
+              payments.
+            </li>
+            <li>Once you have scheduled payments, the module (source code here)
+              triggers the payments at the appointed time.
+            </li>
+            <li>Creating the safe and enabling the module requires a one-time gas
+              fee.
+            </li>
+          </ul>
+          {{! TODO: Gas cost should be handled on this component or be a param ? }}
+          <b>Estimated gas cost: {{gasCost}}</b>
+          {{#if hasEnoughGas}}
+            <div class='safe-setup-modal__section-wallet-info'>
+              {{svgJar
+                'icon-check-circle-ht'
+                class='safe-setup-modal__section-icon'
+              }}
+              <p>
+                Your wallet has sufficient funds to cover the estimated gas cost.
+              </p>
+            </div>
+          {{/if}}
+          {{! TODO: What's the behavior for not having enough gas ? }}
+        </Section>
+        <ActionChin @state={{'default'}}>
+          <:default as |a|>
+            {{! TODO: Provisioning should be handled on this component or be a param to be handled on a diff controller ? }}
+            <a.ActionButton @loading={{@loading}}>
+              Provision
+            </a.ActionButton>
+            <a.CancelButton {{on 'click' @onClose}} @disabled={{@loading}}>
```

Comment:
Why disable this over removing it from the UI when loading?

Reply:
I guess i didn't give much thought to it  😅, hiding it, seems more user friendly.